components/server/ee/src/gitlab/gitlab-app-support.ts

```diff
@@ -51,7 +52,11 @@ export class GitLabAppSupport {
         // we are listing only those projects with access level of maintainers.
         // also cf. https://docs.gitlab.com/ee/api/members.html#valid-access-levels
         //
-        const projectsWithAccess = await api.Projects.all({ min_access_level: "40", perPage: 100 });
+        const projectsWithAccess = await api.Projects.all({
+            min_access_level: "40",
+            perPage: 100,
+            pagination: "keyset",
```

Comment:
Hi @gtsiolis! Would you like to test this change on the preview environment? Maybe you know how to find the 101 repo in the GitLab org/group?

Reply:
@AlexTugarev I don't have maintainer access to such a long list of repositories to try this but took 10 minutes to create a list of 101 repositories via command line in a reproduction group.

However, I can already find the 101th repository in Gitpod.io, see screenshot below. 💭 

For this preview environment, when I add the GitLab integration I'm stuck in the fetching repositories screen with the following error in console.

```js
Error: Request getProviderRepositoriesForUser failed with message: Response code 405 (Method Not Allowed)
```

| Gitpod.io | This preview environment |
|-|-|
| <img width="1440" alt="Screenshot 2022-09-16 at 2 27 13 PM" src="https://user-images.githubusercontent.com/120486/190628983-2cb117fa-6e20-4c8d-bc22-43aff29e3ab1.png"> | <img width="1440" alt="Screenshot 2022-09-16 at 2 24 35 PM" src="https://user-images.githubusercontent.com/120486/190628993-cddec49e-7df0-4df2-b2fd-0ebdbf445f60.png"> |

Reply:
@gtsiolis, in the meantime, I was able to create a test script to create ~200 test projects in https://gitlab.com/alex-gp-test-group?page=1 and wrote a unit test for the request which was expected to be broken. not to forget, those tests are not automated, as they require a personal OAuth token in the environ. 

That said, I'd be more than happy if you could retest after next build is deployed to preview. 🙏🏻

Reply:
I'll make a screenshot to the description as well, once I could verify it myself.

Reply:
> ... in the meantime, I was able to create a test script to create ~200 test projects ... 

🙂  

> ... wrote a unit test for the request which was expected to be broken. not to forget, those tests are not automated, as they require a personal OAuth token ...

It could be nice if we could use a dedicated group for such integrations tests, right? Cross-posting a relevant comment from https://github.com/gitpod-io/gitpod/pull/5362#discussion_r695079714 for visibility.

> ... if you could retest after next build is deployed to preview

I'll keep an eye on the build[[1](https://werft.gitpod-dev.com/job/gitpod-build-at-gl.6)].

> I'll make a screenshot to the description as well

Thank you! 🙇

Reply:
@AlexTugarev I tested this again and works as expected. I can see the 101th repository. ✔️

Reply:
@gtsiolis, see the updated description. Also, sorry to have you seen the broken state in between. 🤦🏻‍♂️